### PR TITLE
[addons] move addon system settings to regular settings window

### PR DESCRIPTION
**Pull request by tamland**
This moves addon related settings to one common area and will allow more settings to added in the future (and without requiring skin changes). Currently it moves notifications/auto update settings from side menu, and, 'administrative' functionality from the 'system' directory.

Todo:
Something to 'ease' the transition for user? (e.g a link to settings from addon browser. Unfortunately it doesn't seem possibly programmically open a settings category atm.)

### CODE REVIEW

**Comment by da-anda** [comment]
I only had a quick look. A few things I noticed:
- new settings IMO need a description, now that they are visible in GUI
- settings labels should get a contextual comment so that translators get an idea of their use case. These start with `#.`
- I think you have to remove the related labels of the removed skin buttons as well
- IMO it would be nicer if the add-on settings would actually be inside the "add-ons" settings category which currently triggers the add-on manager, and that the manager would be integrated differently. But that's a longer story.

**Comment by tamland** [comment]
Just saving the most fun parts for last...

>I think you have to remove the related labels of the removed skin buttons as well

It's for backwards compatibility, or have we just given up on not breaking skin from release to release at this point?

**Comment by ronie** [comment]
we normally break skins with every release. the only ever exception was helix to isengard.
for K*** it will be business as usual, we've already merged one or two PR's that contain BW compatible breaking changes.

that being said, we should always be very careful if we want to remove strings from the language file.
those strings might be re-used by skins in other areas as well or by python addons.

**Comment by Montellese** [comment]
If you want to be able to open a specific settings category in a pre-chosen section you could extend `CGUIWindowSettingsCategory`'s handling of the `GUI_MSG_WINDOW_INIT` message to be able to pass an additional parameter.

**Comment by MartijnKaijser** [comment]
@tamland if @jjd-uk has his settings reshuffle finished most skins will be broken any way

**Comment by tamland** [comment]
Thanks @Montellese. Added (feels a bit hacky though. for some reason it always shows the settings of the first category no matter what m_iCategory is so I had to resort to calling `CreateSettings` again)

**Comment by phil65** [comment]
Thx for this PR, much nicer indeed.
Small suggestion: move settings shortcut into sidebar instead of into VFS.

**Comment by tamland** [comment]
> Small suggestion: move settings shortcut into sidebar instead of into VFS.

Because.. it's better?

**Comment by Montellese** [comment]
@tamland: IIRC I had the same issue in `CGUIWindowSettingsCategory` when I tried to do something similar in one of my branches. Let me check if I can find it again.

**Comment by phil65** [comment]
Because the settings were part of the sidebar before and it's more consistent with other areas in my opinion. As said, just a suggestion, if you disagree then feel free to ignore this.

**Comment by tamland** [comment]
>Because the settings were part of the sidebar before and it's more consistent with other areas in my opinion. As said, just a suggestion, if you disagree then feel free to ignore this.

Will do. I'm just trying to drag out some information out of you. I don't have any opinion either way.

**Comment by Montellese** [comment]
@tamland: I've looked into this again and I'll send you a PR that should be better.

**Comment by Montellese** [comment]
@tamland: Without 8804fb7107569138bac99905f1e213350bfd28b4 this PR doesn't build with VS2013.

6e0ff473dcecda22f88c3ba33f69b82cb41194d6 gets rid of the manual call to `CreateSettings()` and of the manual `for` loop. Not setting `m_iCategory` is the way to go because then `CreateSettings()` is called by the logic in `CGUIDialogSettingsBase` which handles `GUI_MSG_FOCUSED`.

**Comment by tamland** [comment]
@Montellese Thanks! Much better

**Comment by tamland** [comment]
Could someone provide xcode sync?

**Comment by mkortstiege** [comment]
> Could someone provide xcode sync?

Please pick https://github.com/mkortstiege/xbmc/commit/9ad693328911a31298043a637ca197faac4b2518

**Comment by tamland** [comment]
Thanks! jenkins build this please

File: addons/skin.confluence/720p/AddonBrowser.xml [hunk 1]
```diff
@@ -108,6 +97,11 @@
 					<include>ButtonCommonValues</include>
 					<label>24034</label>
 				</control>
+				<control type="button" id="5">
+					<description>Settings foreign</description>
+					<include>ButtonCommonValues</include>
+					<label>5</label>
```
Comment (MartijnKaijser) [inline addons/skin.confluence/720p/AddonBrowser.xml]: could you use string id 10004 to get rid of low number usage? Is the description correct?

Comment (tamland) [inline addons/skin.confluence/720p/AddonBrowser.xml]: Thanks. Sloppy copy-paste;)

**Pull request merged**